libs/datasets/timeseries.py

```diff
@@ -782,6 +782,39 @@ def add_new_cases(timeseries: MultiRegionTimeseriesDataset) -> MultiRegionTimese
     return new_timeseries
 
 
+def _calculate_modified_zscore(series: pd.Series, window: int = 10) -> pd.Series:
+    """Calculates zscore removing zeros"""
+    series = series.copy()
+    series[series == 0] = None
+    rolling_series = series.rolling(window=window, min_periods=3)
+    # Shifting one to exclude current datapoint
+    mean = rolling_series.mean().shift(1)
+    std = rolling_series.std(ddof=0).shift(1)
+    z = (series - mean) / std
```

Comment:
What happens when std = 0, how do we handle the inf downstream?

Reply:
hahah i bet there is never a timeseries that consistent

Reply:
but it's a good point i'll look into it